core/node/node_framework/src/resource/resource_collection.rs

```diff
@@ -104,3 +104,64 @@ impl<T: Resource + Clone> ResourceCollection<T> {
         (*handle).clone()
     }
 }
+
+#[cfg(test)]
+mod tests {
+    use std::time::Duration;
+
+    use tokio::time::timeout;
+
+    use super::*;
+
+    #[derive(Debug, Clone, PartialEq)]
+    struct TestResource(Arc<u8>);
+
+    impl Resource for TestResource {
+        fn resource_id() -> ResourceId {
+            ResourceId::new("test_resource")
+        }
+    }
+
+    #[tokio::test]
+    async fn test_push() {
+        let collection = ResourceCollection::<TestResource>::new();
+        let resource = TestResource(Arc::new(1));
+
+        collection.clone().push(resource).unwrap();
+
+        assert_eq!(
+            *collection.resources.lock().unwrap(),
+            vec![TestResource(Arc::new(1))]
+        );
+    }
+
+    #[tokio::test]
+    async fn test_already_wired() {
+        let mut collection = ResourceCollection::<TestResource>::new();
+        let resource = TestResource(Arc::new(1));
+
+        let rc_clone = collection.clone();
+
+        collection.on_resource_wired();
+
+        assert_eq!(
+            rc_clone.push(resource),
+            Err(ResourceCollectionError::AlreadyWired)
+        );
+    }
+
+    #[tokio::test]
+    async fn test_resolve() {
+        let mut collection = ResourceCollection::<TestResource>::new();
+
+        collection.on_resource_wired();
+
+        let resolved = collection.resolve().await;
+        assert_eq!(resolved, vec![]);
+
+        // check the case that it is not allowed to resolve the collection when it is not wired
+        let collection = ResourceCollection::<TestResource>::new();
+        let result = timeout(Duration::from_secs(1), collection.resolve()).await;
```

Comment:
I think 50ms would be sufficient for the purpose of this test. 1 second is a lot of time, and we don't want to make the tests longer than they should be.

Reply:
Also I think it makes sense to extend this test by calling `on_resource_wired()` and making sure that it can be resolved after that.